### ISSUE TITLE
EmuDeck does not work on winesapOS

### ISSUE THREAD

**Issue by vyktorjonas** [LukeShortCloud/winesapOS#636]
> I know I just opened another issue but since it's a different topic and there's 0 info about winesapOS online, I thought this would be helpful for others.

Since WinesapOS sells itself as a steamdeck for everyone, I was planning on using extensions EmuDeck and DeckyLoader with it, the .desktop file works and it even installs to my surprise, thing is, there's no integration with the installed Steam app whatsoever, so it doesn't matter what extension you install, it won't alter your Steam app at all, would be very useful if that could be looked at for future updates

**Comment by LukeShortCloud** [comment]
Hey @vyktorjonas thanks for the suggestion! I want to make sure I understand this correctly: (1) EmuDeck doesn't add emulators and games to as non-Steam games to Steam and (2) DeckyLoader doesn't load plugins into Steam? I'm unsure as to why those would not work on winesapOS. I'll look into it.

**Comment by vyktorjonas** [comment]
EmuDeck was the only one I tried, I didn't try decky loader since the problem seemed to be systemic
EmuDeck does not add emulators and cannot launch the ROMs manager, even though it claims to have installed the emulators, I believe it happens because the extension dumps the installed files in another folder that only steamdeck uses but winesap doesn't, just a hunch though

**Comment by LukeShortCloud** [comment]
Thanks for the additional information, that's really helpful. If I find anything out, I'll let you know.

**Comment by LukeShortCloud** [comment]
I did a bit more investigation into this. It appears that there are some hard-coded assumptions about where directories are.

```
  "environmentVariables": {
    "steamDirectory": "/home/deck/.steam/steam",
    "romsDirectory": "/run/media/mmcblk0p1/Emulation/roms",
```

https://github.com/dragoonDorise/EmuDeck/blob/2.1.0/configs/steam-rom-manager/userData/userSettings.json#L11

To workaround this, create a user named `deck`. The directory `/run/media/mmcblk0p1/Emulation/roms/` also needs to be symlinked to another location (`/home/deck/roms/`, for example).

I see a lot of reports online that configuring EmuDeck is difficult. For example:

https://www.reddit.com/r/SteamDeck/comments/wr9kys/change_rom_folder_on_emu_deck/

The best I can do is provide some documentation and tips for how to get EmuDeck working on winesapOS (or any other Linux distribution, for that matter).

**Comment by LukeShortCloud** [comment]
@vyktorjonas Try running these commands and then re-install EmuDeck again. Does that help?

```
sudo ln -s /home/winesap /home/deck
mkdir -p ~/Emulation/roms
sudo mkdir -p /run/media/mmcblk0p1/
sudo ln -s ~/Emulation /run/media/mmcblk0p1/Emulation
```

**Comment by LukeShortCloud** [comment]
I did some testing. Here are my results.

Symlinking the `deck` home directory to `winesap` allows the emulators to be installed. So does creating the `deck` user and installing EmuDeck with that user.

I am running into problems with opening the Steam ROM Manager. It always asks if I want to exit Steam first (even if Steam is not even running). The exact message is: `Exit Steam to launch Steam Rom Manager?`. I select "Yes" and then nothing happens. I tried to uninstall `appimagelauncher` to see if that was causing some weird issues. It did not help or change anything.

**Comment by LukeShortCloud** [comment]
There was a report from 1.5 months ago where someone had the same issue of Steam Rom Manager not launching. The only vague comment about it was that there might be some hints in the wiki. I checked the wiki for EmuDeck and, separately, for Steam Rom Manager. I could not find anything helpful at a first glance.

`https://github.com/dragoonDorise/EmuDeck/issues/823`

**Comment by LukeShortCloud** [comment]
All of the symlinks seem to be unnecessary. EmuDeck has been updated sometime in the last year to use `${HOME}` and `${USER}` variables instead of hard-coding to `deck` in most cases (and where there is hard-coding, it seems to do  string substitutions).

I retested many times on a clean installation of winesapOS and consistently see a `curl` error in the EmuDeck installer log. That means that parts of EmuDeck are failing to download. I opened up an upstream GitHub Issue to see if their maintainers can help us out: https://github.com/dragoonDorise/EmuDeck/issues/849

**Comment by LukeShortCloud** [comment]
@vyktorjonas We solved the problem! Install this missing system dependency before re-installing EmuDeck:

```
sudo pacman -S jq
```

This will be addressed in upcoming versions of winesapOS and EmuDeck.

**Comment by LukeShortCloud** [comment]
I have added full support for EmuDeck in winesapOS. As a bonus, EmuDeck will be included in wineesapOS 3.4.0 in the performance and secure images (with an option to install it on the minimal image).

**Comment by LukeShortCloud** [comment]
@vyktorjonas winesapOS 3.4.0 is out now and includes a desktop shortcut to install EmuDeck along with all of the dependencies it needs. Thanks again for your report!

https://github.com/LukeShortCloud/winesapOS/releases/tag/3.4.0
